File: student-management-system/src/main/java/com/student/studentmanagementsystem/service/StudentService.java
package com.student.studentmanagementsystem.service;

import java.util.List;

import com.student.studentmanagementsystem.entities.Student;
import com.student.studentmanagementsystem.entities.StudentResponse;

public interface StudentService {
    List<Student> getAllStudents();
    StudentResponse getStudentById(Long student_id);
    Student saveStudent(Student student);
    Student updateStudent(Long student_id, Student updatedStudent);
    void deleteStudent(Long student_id);
}

File: student-management-system-frontend/src/app/services/student.service.ts
// student.service.ts

import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../environments/environment';
import { Student } from '../model/student.model';

@Injectable({
  providedIn: 'root',
})
export class StudentService {
  
  // private apiUrl = 'http://localhost:8080/api/students';
  private apiUrl = `${environment.apiUrl}/students`;

  constructor(private http: HttpClient) {}

  getAllStudents(): Observable<Student[]> {
    return this.http.get<Student[]>(this.apiUrl);
  }

  getStudentById(id: number): Observable<Student> {
    return this.http.get<Student>(`${this.apiUrl}/${id}`);
  }

  saveStudent(student: Student): Observable<Student> {
    return this.http.post<Student>(this.apiUrl, student);
  }

  updateStudent(id: number, student: Student): Observable<Student> {
    return this.http.put<Student>(`${this.apiUrl}/${id}`, student);
  }

  addStudent(student: Student): Observable<void> {
    return this.http.post<void>(this.apiUrl, student);
  }

  deleteStudent(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}
